### ISSUE TITLE
Duplicate entry error in database

### ISSUE THREAD

**Issue by thepurpleblob** [ndunand/moodle-tool_mergeusers#107]
Running the latest master on 2.9.3...

Merged «ltutest2» (user ID = 49) into «ltutest3» (user ID = 50)

For further reference, these results are recorded in the log id 1.
Some error occurred:

Exception thrown when merging: 'Error writing to database".
Duplicate entry '721-50-0-0' for key 'mdl_assisubm_assusegroatt_uix'
Trace:
#0 /var/www/moodle29/lib/dml/mysqli_native_moodle_database.php(974): moodle_database->query_end(false)
#1 /var/www/moodle29/admin/tool/mergeusers/lib/table/generictablemerger.php(225): mysqli_native_moodle_database->execute('UPDATE mdl_assi...')
#2 /var/www/moodle29/admin/tool/mergeusers/lib/table/generictablemerger.php(80): GenericTableMerger->updateRecords(Array, Array, 'userid', Array, Array)
#3 /var/www/moodle29/admin/tool/mergeusers/lib/mergeusertool.php(281): GenericTableMerger->merge(Array, Array, Array)
#4 /var/www/moodle29/admin/tool/mergeusers/lib/mergeusertool.php(216): MergeUserTool->_merge('50', '49')
#5 /var/www/moodle29/admin/tool/mergeusers/index.php(135): MergeUserTool->merge('50', '49')
#6 {main}

**Comment by jpahullo** [comment]
Hi!

Thanks for reporting! It is very important for this plugin and for us to know about misbehaviours and problems with it.

Try adding this configuration into "compoundindexes" associative array in your "config.local.php":

```
       'assign_submission' => array( // unique key mdl_assisubm_assusegroatt_uix
            'userfield' => array('userid'),
            'otherfields' => array('assignment', 'groupid', 'attemptnumber'),
        ),
```

if you have no config.local.php file, just create it into config/config.local.php and place this text:

```
return array(
    // Local list of compound indexes.
    // Table names must be without $CFG->prefix.
    'compoundindexes' => array(
       'assign_submission' => array( // unique key mdl_assisubm_assusegroatt_uix
            'userfield' => array('userid'),
            'otherfields' => array('assignment', 'groupid', 'attemptnumber'),
        ),
    ),
);
```

This should work for you. Let us know. Thanks a lot in advance!

Jordi

**Comment by ndunand** [comment]
@thepurpleblob could you let us know whether this solved it? Thanks!

**Comment by ndunand** [comment]
Closed as fixed by 3e5d08d
